lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -443,7 +443,7 @@ private ColumnChunk readPrimitive(PrimitiveField field)
         ColumnChunk columnChunk = columnReader.readPrimitive();
 
         // update max size per primitive column chunk
-        double bytesPerCell = ((double) columnChunk.getBlock().getSizeInBytes()) / batchSize;
+        double bytesPerCell = ((double) columnChunk.getMaxBlockSize()) / batchSize;
```

Comment:
How is this related?

Reply:
We wanted to avoid calling `DictionaryBlock#getSizeInBytes` (it's an expensive method). So we compute an approximate upper bound for dictionary block size in `DictionaryValuesBuffer#getMaxDictionaryBlockSize` and populate that in `ColumnChunk` for usage here. For non-dictionary blocks, this continues to return result of `Block#getSizeInBytes`.